### PR TITLE
[WIP] Add properties to axis

### PR DESCRIPTION
**Pull request by YuhanLu**
#1158 addedadded properties to axis and matching test cases, docs not finished yet  @kanitw

### CODE REVIEW

File: src/compile/axis.ts [hunk 1]
```diff
@@ -186,6 +186,14 @@ export function tickSize(model: Model, channel: Channel) {
   return undefined;
 }
 
+export function tickSizeEnd(model: Model, channel: Channel) {
```
Comment (kanitw) [inline src/compile/axis.ts]: Please also move `tickSizeEnd` above from the group below
`    // b) properties without rules, only produce default values in the schema, or explicit value if specified
`

to 

`    // a) properties with special rules (so it has axis[property] methods) -- call rule functions
`

See around line 72.  (We sort each group by alphabetical order)

File: src/compile/axis.ts [hunk 2]
```diff
@@ -210,18 +218,42 @@ export function title(model: Model, channel: Channel) {
   return maxLength ? truncate(fieldTitle, maxLength) : fieldTitle;
```
Comment (kanitw) [inline src/compile/axis.ts]: Same as `tickSizeEnd`

File: src/compile/axis.ts [hunk 3]
```diff
@@ -210,18 +218,42 @@ export function title(model: Model, channel: Channel) {
...
 }
 
+export function titleOffset(model: Model, channel: Channel) {
```
Comment (kanitw) [inline src/compile/axis.ts]: Same as `tickSizeEnd`

File: src/compile/axis.ts [hunk 4]
```diff
@@ -210,18 +218,42 @@ export function title(model: Model, channel: Channel) {
...
+    }
+    return undefined;
+}
+
 export namespace properties {
   export function axis(model: Model, channel: Channel, axisPropsSpec, def) {
     const axis = model.axis(channel);
 
     return extend(
+      axis.axisColor !== undefined ?
```
Comment (kanitw) [inline src/compile/axis.ts]: Please use consistent tabbing style

File: src/compile/axis.ts [hunk 5]
```diff
@@ -210,18 +218,42 @@ export function title(model: Model, channel: Channel) {
...
+      {stroke: {value: axis.axisColor}} : {},
       axis.axisWidth !== undefined ?
-        { strokeWidth: {value: axis.axisWidth} } :
-        {},
+        {strokeWidth: {value: axis.axisWidth }} : {},
+
+      axis.tickColor !== undefined ? { ticks: {stroke : {value: axis.tickColor} } } : {},
```
Comment (kanitw) [inline src/compile/axis.ts]: Does this really work correctly?  This looks incorrect.

**Comment by kanitw** [comment]
Please also rebase to master :)

File: src/compile/axis.ts (hunk 1 above)
Comment (YuhanLu) [inline src/compile/axis.ts]: Do you mean we need to group them in to a) and b) that all the functions in a) are above b) and in alphabetical order?

Comment (kanitw) [inline src/compile/axis.ts]: I mean move `'tickSizeEnd'` from line 75 to line 73 and keep them in alphabetical order.  No need to move the whole method.  :)

**Pull request closed without merging**